src/libsyncengine/update_detection/file_system_observer/remotefilesystemobserverworker.cpp

```diff
@@ -482,7 +482,7 @@ ExitCode RemoteFileSystemObserverWorker::sendLongPoll(bool &changes) {
 ExitCode RemoteFileSystemObserverWorker::processActions(Poco::JSON::Array::Ptr actionArray) {
     if (!actionArray) return ExitCodeOk;
 
-    std::set<NodeId, std::equal_to<>> movedItems;
+    std::set<NodeId, std::less<>> movedItems;
```

Comment:
As `std::less` is the default comparator, could we simply use `std:: set<NodeId>`? Unless you see any advantage to explicitly precise the comparator in this case?

Reply:
![image](https://github.com/Infomaniak/desktop-kDrive/assets/82284536/00b8c129-0e17-4c26-a0a8-d05eec9b4c79)

Reply:
'std::less<Key>' is the default while Sonar suggest to use 'std:less<>'.

> std::less<> is transparent, std::less<anything> is not

https://rules.sonarsource.com/cpp/RSPEC-6021/